sdk/communication/AzureCommunicationChat/Source/PushNotification/PushNotificationClient.swift

```diff
@@ -0,0 +1,217 @@
+                        self.aesKey = "0000000000000000B00000000000000000000000AES="
+                        self.authKey = "0000000000000000B0000000000000000000000AUTH="
+                    } else {
+                        self.aesKey = firstKey
+                        self.authKey = secondKey
+                    }
+
+                    // Create RegistrarClientDescription (It should match valid APNS templates)
+                    let clientDescription = RegistrarClientDescription(
+                        aesKey: self.aesKey,
+                        authKey: self.authKey,
+                        cryptoMethod: PushNotificationClient.cryptoMethod
+                    )
+
+                    // Create RegistrarTransportSettings (Path is device token)
+                    let transport = RegistrarTransportSettings(
+                        path: self.deviceRegistrationToken
+                    )
+
+                    // Register for push notifications
+                    self.registrarClient = createdRegistrarClient
+
+                    guard let registrarClient = self.registrarClient else {
+                        completionHandler(.failure(
+                            AzureError
+                                .client("Failed to start push notifications. RegistrarClient is nil.")
+                        ))
+                        return
+                    }
+
+                    registrarClient.setRegistration(with: clientDescription, for: [transport]) { result in
+                        switch result {
+                        case let .success(response):
+                            self.pushNotificationsStarted = true
+                            completionHandler(.success(response))
+                        case let .failure(error):
+                            self.options.logger
+                                .error("Failed to start push notifications with error: \(error.localizedDescription)")
+                            completionHandler(.failure(AzureError.client("Failed to start push notifications", error)))
+                        }
+                    }
+                case let .failure(error):
+                    completionHandler(.failure(AzureError.client("Failed to initialize the RegistrarClient.", error)))
+                }
+            }
+        )
+    }
+
+    internal func stopPushNotifications(
+        completionHandler: @escaping (Result<HTTPResponse?, AzureError>) -> Void
+    ) {
+        // Report an error if registrarClient doesn't exist
+        if registrarClient == nil {
+            completionHandler(.failure(
+                AzureError
+                    .client(
+                        "RegistrarClient is not initialized, cannot stop push notificaitons. Ensure startPushNotifications() is called first."
+                    )
+            ))
+        } else {
+            // Unregister for Push Notifications
+            registrarClient!.deleteRegistration { result in
```

Comment:
Do not use force-unwrap in production code.

Reply:
Fixed